### PR TITLE
Fixes needed to run drivers

### PR DESCRIPTION
**Pull request by learn-more**
- Fix QlOsUtils X8664 MDL generation
- Fix passthru smashing the stack
- Fix ntoskrnl_IoCreateDevice when called from IoCreateDevice

### CODE REVIEW

File: qiling/os/utils.py [hunk 1]
```diff
@@ -122,10 +122,7 @@ def normalize(path):
 class QlOsUtils:
     def __init__(self, ql):
         self.ql = ql
-        self.archtype = None
-        self.ostype = None
         self.path = None
-        self.archendian = None
```
Comment (learn-more) [inline qiling/os/utils.py:128]: These were never initialized, so I removed them.

File: qiling/os/windows/dlls/ntoskrnl.py [hunk 2]
```diff
@@ -179,7 +179,7 @@ def ntoskrnl_IoCreateDevice(ql, address, params):
     device_object.CurrentIrp.value = 0
     device_object.Timer.value = 0
     device_object.Flags = 0x00000080  # DO_DEVICE_INITIALIZING
-    if params['Exclusive']:
+    if 'Exclusive' in params and params['Exclusive']:
```
Comment (learn-more) [inline qiling/os/windows/dlls/ntoskrnl.py:182]: Exclusive was not passed in params when this was called from `IoCreateDevice`

File: qiling/os/os.py [hunk 3]
```diff
@@ -304,18 +304,18 @@ def _call_api(self, name, params, result, address, return_address):
...
 
         # append syscall to list
         self._call_api(func.__name__, params, result, self.ql.reg.arch_pc, ret_addr)
 
-        # update stack pointer
-        self.ql.reg.arch_sp = self.ql.reg.arch_sp + ((param_num + 1) * 4)
-
         if not passthru and self.PE_RUN:
+            # update stack pointer
+            self.ql.reg.arch_sp = self.ql.reg.arch_sp + ((param_num + 1) * 4)
```
Comment (learn-more) [inline qiling/os/os.py:317]: If the original binary is handling the implementation (passthru), the original function should also clean up the stack.

File: qiling/os/windows/dlls/ntoskrnl.py (hunk 2 above)
Comment (nullableVoidPtr) [reply qiling/os/windows/dlls/ntoskrnl.py:182]: Can't this be `params.get('Exclusive') is not None`?

File: qiling/os/utils.py (hunk 1 above)
Comment (wtdcode) [reply qiling/os/utils.py:128]: No problem.

**Comment by wtdcode** [comment]
Thanks for your PR. Could you add some tiny tests?

File: qiling/os/windows/dlls/ntoskrnl.py (hunk 2 above)
Comment (learn-more) [reply qiling/os/windows/dlls/ntoskrnl.py:182]: That will not handle the case of Exclusive being `False` correctly

**Comment by learn-more** [comment]
> Thanks for your PR. Could you add some tiny tests?

I asked about this in the telegram group, and it was suggested there to put this in a separate PR.

That PR will be created when #397 is merged, since that functionality is needed.

**Pull request merged**